core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptVersionConfiguration.java

```diff
@@ -85,24 +90,40 @@ private ScriptVersionConfiguration() {
         return scriptVersions;
     }
 
+    /*@Override
+    public void delete(ScriptVersionKey scriptVersionKey) {
+        *//*LOGGER.trace(MessageFormat.format("deleting ScriptVersion {0}", scriptVersionKey.toString()));
+        String deleteStatement = deleteStatement(scriptVersionKey);
+        getMetadataRepository().executeUpdate(deleteStatement);*//*
+        throw new UnsupportedOperationException();
+    }*/
+
     @Override
     public void delete(ScriptVersionKey scriptVersionKey) {
         LOGGER.trace(MessageFormat.format("deleting ScriptVersion {0}", scriptVersionKey.toString()));
-        String deleteStatement = deleteStatement(scriptVersionKey);
+        String deleted_At = getCurrentTimeStampAsString();
+        String deleteStatement = markInactiveScriptVersion(scriptVersionKey,deleted_At);
         getMetadataRepository().executeUpdate(deleteStatement);
     }
 
+    public String getCurrentTimeStampAsString(){
+        SimpleDateFormat sdf1 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        return sdf1.format(timestamp);
+    }
+
```

Comment:
have a look at the SQLTools of the iesi-core project and replace this logic with the logic implemented there